projects/cm_mcu/FireFlyTask.c

```diff
@@ -373,7 +373,7 @@ static int read_ff_register(const char *name, uint8_t reg_addr, uint8_t *value,
 
     if (! res ) {
     // Read from register.
-    res = apollo_i2c_ctl_reg_r(i2c_device, ff_i2c_addrs[ff].dev_addr, reg_addr, size, value);
+    res = apollo_i2c_ctl_reg_r(i2c_device, ff_i2c_addrs[ff].dev_addr, 1, (uint16_t)packed_reg_addr, size, *((uint32_t*)value));
```

Comment:
do you need a cast here for `packed_reg_addr`? I think it's already `uint16_t`.

Reply:
fixed it.